packages/patternfly-4/react-core/src/components/ChipGroup/Chip.tsx

```diff
@@ -1,25 +1,64 @@
-import React from 'react';
+import * as React from 'react';
 import { css } from '@patternfly/react-styles';
-import PropTypes from 'prop-types';
-import ChipButton from './ChipButton';
+import { ChipButton } from './ChipButton';
+import { OneOf } from '../../helpers/typeUtils';
 import { Tooltip, TooltipPosition } from '../Tooltip';
 import { TimesCircleIcon } from '@patternfly/react-icons';
 import styles from '@patternfly/patternfly/components/Chip/chip.css';
 import GenerateId from '../../helpers/GenerateId/GenerateId';
-import { componentShape } from '../../helpers/componentShape';
 
-class Chip extends React.Component {
-  span = React.createRef();
-  state = { isTooltipVisible: false };
+export interface ChipProps extends React.HTMLProps<HTMLDivElement> {
+  /** Content rendered inside the chip text */
+  children?: React.ReactNode;
+  /** Aria Label for close button */
+  closeBtnAriaLabel?: string;
+  /** Additional classes added to the chip item */
+  className?: string; 
+  /** Flag indicating if the chip has overflow */
+  isOverflowChip?: boolean;
+  /** Flag if chip is read only */
+  isReadOnly?: boolean
+  /** Function that is called when clicking on the chip button */
+  onClick?(event: React.MouseEvent): void
```

Comment:
Nitpicky, but please use arrow function syntax. Change to `onClick?: (event: React.MouseEvent) => void`